dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,561 @@
+{
+    if (const auto& item = args.Element().try_as<winrt::BreadcrumbItem>())
+    {
+        if (const auto& itemImpl = winrt::get_self<BreadcrumbItem>(item))
+        {
+            // The first element is always the ellipsis item
+            itemImpl->SetItemsRepeater(*this);
+
+            const uint32_t itemIndex = args.Index();
+            if (itemIndex == 0)
+            {
+                itemImpl->SetPropertiesForEllipsisNode();
+                itemImpl->SetFlyoutDataTemplate(DropdownItemTemplate());
+
+                m_ellipsisBreadcrumbItem.set(item);
+            }
+            else
+            {
+                const auto& itemsSourceAsList = this->ItemsSource().as<winrt::Collections::IVector<winrt::IInspectable>>();
+                const uint32_t itemCount = itemsSourceAsList.Size();
+
+                if (itemIndex == itemCount)
+                {
+                    UpdateLastElement(item);
+                }
+                else
+                {
+                    // Any other element just resets the visual properties
+                    itemImpl->ResetVisualProperties();
+                }
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnElementIndexChangedEvent(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementIndexChangedEventArgs& args)
+{
+    if (auto const element = args.Element())
+    {
+        element.SetValue(winrt::AutomationProperties::PositionInSetProperty(), box_value(args.NewIndex() + 1));
+
+        // When the selected item's index changes, update selection to match
+        if (auto const elementAsToggle = element.try_as<winrt::ToggleButton>())
+        {
+            if (SharedHelpers::IsTrue(elementAsToggle.IsChecked()))
+            {
+                FocusElement(args.NewIndex());
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnElementClearingEvent(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementClearingEventArgs& args)
+{
+    if (const auto& item = args.Element().try_as<winrt::BreadcrumbItem>())
+    {
+        const auto& itemImpl = winrt::get_self<BreadcrumbItem>(item);
+        itemImpl->ResetVisualProperties();
+        itemImpl->RevokeListeners();
+    }
+}
+
+winrt::IInspectable Breadcrumb::GenerateInternalItemsSource()
+{
+    // An internal ItemsSource is generated internally to add the
+    // ellipsis button
+    const auto& newItemsSource = winrt::make<Vector<IInspectable>>();
+
+    // The ellipsis button is created as a BreadcrumbItem, the element
+    // factory verifies that this element is already a BreadcrumbItem
+    // so no extra processing is needed
+    const auto& ellipsisItem = winrt::make<BreadcrumbItem>();
+    ellipsisItem.Content(winrt::box_value(L"..."));
+    newItemsSource.Append(ellipsisItem);
+
+    if (const auto& itemsList = ItemsSource().try_as<winrt::Foundation::Collections::IIterable<winrt::IInspectable>>())
```

Comment:
As discussed, it would be better to create a collection wrapper that does not clone the original source, but just accesses the original items.